### PR TITLE
Various QoL improvements

### PR DESCRIPTION
**Pull request by TacoTheDank**
### 1st commit
- Deleted unnecessary files

### 2nd commit
- Updated gradle wrapper to 6.7.1 using git bash

### 3rd commit
- Reorganize build gradles
- Remove unneeded stuff in engine module
- Organize libraries
- Remove unused easypermissions library

### 4th commit
- Update buildTools 29.0.3 -> 30.0.3
- Update ndkVersion 21.3 -> 22.0 ([changelog](https://github.com/android/ndk/wiki/Changelog-r22))
- Update AppCompat 1.1.0 -> 1.2.0 ([changelog](https://developer.android.com/jetpack/androidx/releases/appcompat))
- Update Lifecycle 2.1.0 -> 2.2.0 ([changelog](https://developer.android.com/jetpack/androidx/releases/lifecycle))
- Update transitive Preference 1.0.0 -> 1.1.1 ([changelog](https://developer.android.com/jetpack/androidx/releases/preference))
- Update transitive Google Material 1.0.0 -> 1.2.1 ([changelog](https://github.com/material-components/material-components-android/releases))
- Update Butterknife 10.2.1 -> 10.2.3 ([changelog](https://github.com/JakeWharton/butterknife/blob/master/CHANGELOG.md))
- Update Google Services 4.3.3 -> 4.3.4 ([changelog](https://firebase.google.com/support/release-notes/android#2020-09-30))

### 5th commit
- Migrated results page to ViewPager2 (see [this](https://developer.android.com/training/animation/vp2-migration))

### 6th commit
- Removed stray jetbrains annotations

### 7th commit
- Fixed various deprecations

### 8th commit
- Optimized imports throughout the project

### CODE REVIEW

**Comment by lorenzoPrimi** [comment]
Hi @TacoTheDank thanks for the PR!
Is this ready to review?

**Comment by TacoTheDank** [comment]
@lorenzoPrimi Yes, go right ahead! :)

**Comment by lorenzoPrimi** [comment]
Do you know why the compile is failing on the test?
```
* What went wrong:
Execution failed for task ':app:processStableFullDebugResources'.
> A failure occurred while executing com.android.build.gradle.internal.tasks.Workers$ActionFacade
   > Android resource linking failed
     /Users/runner/work/probe-android/probe-android/app/src/main/res/values/styles.xml:32:2-36:10: AAPT: error: resource color/colorPrimary (aka org.openobservatory.ooniprobe:color/colorPrimary) not found.
         
     /Users/runner/work/probe-android/probe-android/app/src/main/res/values/styles.xml:32:2-36:10: AAPT: error: resource color/colorPrimaryDark (aka org.openobservatory.ooniprobe:color/colorPrimaryDark) not found.
         
     /Users/runner/work/probe-android/probe-android/app/src/main/res/values/styles.xml:32:2-36:10: AAPT: error: resource color/colorAccent (aka org.openobservatory.ooniprobe:color/colorAccent) not found.

```

**Comment by TacoTheDank** [comment]
@lorenzoPrimi I wasn't aware of this 🤔 I looked around and found that these resources were being called from the easypermissions library that wasn't being used, so I'll directly use the colors they were calling.

File: engine/build.gradle [hunk 1]
```diff
@@ -2,36 +2,19 @@ apply plugin: 'com.android.library'
...
-
 }
 
 dependencies {
-    implementation fileTree(dir: 'libs', include: ['*.jar'])
     implementation 'org.ooni:oonimkall:2020.12.07-102305'
-    implementation 'androidx.appcompat:appcompat:1.2.0'
-    testImplementation 'junit:junit:4.12'
-    androidTestImplementation 'androidx.test.ext:junit:1.1.2'
-    androidTestImplementation 'androidx.test.espresso:espresso-core:3.3.0'
```
Comment (lorenzoPrimi) [inline engine/build.gradle:36]: @bassosimone can you check these lines?

Comment (bassosimone) [reply engine/build.gradle:36]: sgtm

**Comment by bassosimone** [comment]
@TacoTheDank I love how you structured your contribution and explained yourself! Thank you so much!

**Comment by lorenzoPrimi** [comment]
Merging this as looks very good to me.
Thank you @TacoTheDank on behalf of the entire OONI Team!

**Comment by TacoTheDank** [comment]
Thank you very much! :heart:

**Pull request merged**